packages/console/src/pages/TenantSettings/tabs/Domains/components/CustomDomain/components/CustomDomainHeader/index.tsx

```diff
@@ -0,0 +1,103 @@
+import { type AdminConsoleKey } from '@logto/phrases';
+import { DomainStatus, type Domain } from '@logto/schemas';
+import { useEffect } from 'react';
+import { toast } from 'react-hot-toast';
+import { Trans, useTranslation } from 'react-i18next';
+
+import Delete from '@/assets/images/delete.svg';
+import More from '@/assets/images/more.svg';
+import ActionMenu, { ActionMenuItem } from '@/components/ActionMenu';
+import CopyToClipboard from '@/components/CopyToClipboard';
+import DynamicT from '@/components/DynamicT';
+import Tag from '@/components/Tag';
+import type { Props as TagProps } from '@/components/Tag';
+import useApi from '@/hooks/use-api';
+import { useConfirmModal } from '@/hooks/use-confirm-modal';
+
+import { isDomainStatus } from '../../utils';
+
+import * as styles from './index.module.scss';
+
+type Props = {
+  customDomain: Domain;
+  onDeleteCustomDomain: () => void;
+};
+
+const domainStatusToTag: Record<
+  DomainStatus,
+  { title: AdminConsoleKey; status: TagProps['status'] }
+> = {
+  [DomainStatus.PendingVerification]: { title: 'domain.status.connecting', status: 'alert' },
+  [DomainStatus.PendingSsl]: { title: 'domain.status.connecting', status: 'alert' },
+  [DomainStatus.Active]: { title: 'domain.status.in_used', status: 'success' },
+  [DomainStatus.Error]: { title: 'domain.status.failed_to_connect', status: 'error' },
+};
+
+function CustomDomainHeader({ customDomain: { id, domain, status }, onDeleteCustomDomain }: Props) {
+  const { t } = useTranslation(undefined, { keyPrefix: 'admin_console' });
+  // TODO @xiaoyijun Remove this type assertion when the LOG-6276 issue is done by @wangsijie
+  const tag = domainStatusToTag[isDomainStatus(status) ? status : DomainStatus.Error];
+  const { show, cancel } = useConfirmModal();
+  const api = useApi();
+
+  useEffect(() => {
+    return () => {
+      cancel();
```

Comment:
What is the use case?

Reply:
When we leave the page, we should close the confirm modal.

Reply:
Updated, since we have the PR to fix this issue: #3981